src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -780,7 +781,7 @@ namespace winrt::TerminalApp::implementation
     void TerminalPage::_DuplicateTabViewItem()
     {
         const int& focusedTabIndex = _GetFocusedTabIndex();
-        const auto& _tab = _tabs.at(focusedTabIndex);
+        const auto& _tab = winrt::get_self<Tab>(_tabs.GetAt(focusedTabIndex));
```

Comment:
Actually, this really should be the following block of code, but I missed this one function so thanks for the catch!
```
winrt::com_ptr<Tab> tab;
tab.copy_from(winrt::get_self<Tab>(_tabs.GetAt(focusedTabIndex));
```
In regards to your question though, since `_tabs` is an `IObservableVector<winrt::TerminalApp::Tab>`, `_tabs.GetAt` isn't sufficient because that would return the projected type of `Tab`, which wouldn't let me call any of `Tab`'s functions. I need to get at its implementation here, so `get_self` would return to me a `implementation::Tab*`. Then I'd need to stuff it in a com_ptr so it doesn't die randomly while this function is doing its thing, hence the `copy_from`.

Reply:
See, I would have thought that a `winrt::com_ptr<Tab>` could be created/copied directly off of another `winrt::TerminalApp::Tab` and not require the `winrt::TerminalApp::implementation::Tab*`.

Reply:
Yeah I went digging for a way to do just that after you told me there might be a way, but I couldn't seem to find it 😢